pkg/operatorsource/catalogsourceconfigcrud_test.go

```diff
@@ -0,0 +1,175 @@
+
+	assert.Error(t, errGot)
+}
+
+// Use Case: Get already existing CatalogSourceConfig object.
+// Expected Result: The existing CatalogSourceConfig object should be returned
+func TestCRUD_Get_CatalogSourceConfig_CscAlreadyExists(t *testing.T) {
+
+	// Set up fakeclient with an existing CSC
+	namespace, name := "marketplace", "foo"
+	labelsWant := map[string]string{
+		"opsrc-group":                           "Community",
+		operatorsource.OpsrcOwnerNameLabel:      "foo",
+		operatorsource.OpsrcOwnerNamespaceLabel: "marketplace",
+	}
+	cscIn := helperNewCatalogSourceConfigWithLabels(namespace, name, labelsWant)
+
+	objs := []runtime.Object{
+		cscIn,
+	}
+
+	scheme := runtime.NewScheme()
+	apis.AddToScheme(scheme)
+
+	fakeclient := fake.NewFakeClientWithScheme(scheme, objs...)
+
+	context := context.TODO()
+	opsrc := helperNewOperatorSourceWithPhase(namespace, name, phase.Configuring)
+	catalogsourceconfigcrud := operatorsource.NewCatalogSourceConfigCRUD(fakeclient)
+
+	cscGot, errGot := catalogsourceconfigcrud.GetCatalogSourceConfig(context, opsrc)
+
+	assert.NoError(t, errGot)
+	assert.Equal(t, cscIn, &cscGot)
+}
+
+// Use Case: Try to Get non existing CatalogSourceConfig object.
+// Expected Result: An error should be returned
+func TestCRUD_Get_CatalogSourceConfig_CscDoesNotExists(t *testing.T) {
+
+	// Set up fakeclient without CSC
+	namespace, name := "marketplace", "foo"
+
+	scheme := runtime.NewScheme()
+	apis.AddToScheme(scheme)
+
+	fakeclient := fake.NewFakeClientWithScheme(scheme)
+
+	context := context.TODO()
+	opsrc := helperNewOperatorSourceWithPhase(namespace, name, phase.Configuring)
+	catalogsourceconfigcrud := operatorsource.NewCatalogSourceConfigCRUD(fakeclient)
+
+	_, errGot := catalogsourceconfigcrud.GetCatalogSourceConfig(context, opsrc)
+
+	assert.Error(t, errGot)
+}
+
+// Use Case: Update already existing CatalogSourceConfig object.
+// Expected Result: The existing CatalogSourceConfig object should be returned
+func TestCRUD_Update_CatalogSourceConfig_CscAlreadyExists(t *testing.T) {
+
+	// Set up fakeclient with an existing CSC
+	manifests := "a,b,c"
+	namespace, name := "marketplace", "foo"
+	labelsWant := map[string]string{
+		"opsrc-group":                           "Community",
+		operatorsource.OpsrcOwnerNameLabel:      "foo",
+		operatorsource.OpsrcOwnerNamespaceLabel: "marketplace",
+	}
+	cscIn := helperNewCatalogSourceConfigWithLabels(namespace, name, labelsWant)
+
+	objs := []runtime.Object{
+		cscIn,
+	}
+
+	scheme := runtime.NewScheme()
+	apis.AddToScheme(scheme)
+
+	fakeclient := fake.NewFakeClientWithScheme(scheme, objs...)
+
+	context := context.TODO()
+	opsrc := helperNewOperatorSourceWithPhase(namespace, name, phase.Configuring)
+	catalogsourceconfigcrud := operatorsource.NewCatalogSourceConfigCRUD(fakeclient)
+
+	errGot := catalogsourceconfigcrud.UpdateCatalogSourceConfig(context, opsrc, *cscIn, manifests)
+
+	assert.NoError(t, errGot)
```

Comment:
I think we need to assert by `getting` the object and comparing against the expected.